src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -44,6 +45,7 @@ public abstract class BrokerClient : IBrokerClient {
         private readonly ICredentialsDecorator _credentials;
         private readonly IConsole _console;
         private readonly IServiceContainer _services;
+        private readonly IRSessionProvider _sessionProvider;
```

Comment:
I'd rather pass `IRSession` directly into `HandleUrlAsync`.

Reply:
I will take that as a refactoring task. There are a bunch of things that I want to cleanup. #4071